server/persistence/mysql/games.go

```diff
@@ -0,0 +1,569 @@
+		if err == sql.ErrNoRows {
+			return model.Game{}, persistence.ErrGameNotFound
+		}
+		return model.Game{}, err
+	}
+
+	curScores, lagScores := populateScores(
+		scoreBlue, scoreRed, scoreGreen,
+		lagScoreBlue, lagScoreRed, lagScoreGreen,
+	)
+
+	players, err := g.getPlayersForGame(p1ID, p2ID, p3ID, p4ID)
+	if err != nil {
+		return model.Game{}, err
+	}
+
+	pc, err := g.getPlayerColors(gID)
+	if err != nil {
+		return model.Game{}, err
+	}
+	addInPopulatedColor(curScores, lagScores, pc)
+
+	// If we've errored here, just ignore it and continue
+	cutCard, _ := model.NewCardFromTinyInt(cutCardInt)
+
+	cribCards := getCribCards(cribCardInts)
+
+	bp, err := getBlockingPlayers(blockingPlayers)
+	if err != nil {
+		return model.Game{}, err
+	}
+
+	h, err := getHands(hands)
+	if err != nil {
+		return model.Game{}, err
+	}
+
+	p, err := getPeggedCards(peggedCards)
+	if err != nil {
+		return model.Game{}, err
+	}
+
+	pas, err := g.getActions(gID, int(numActions))
+	if err != nil {
+		return model.Game{}, err
+	}
+
+	game := model.Game{
+		ID:              gID,
+		CurrentScores:   curScores,
+		LagScores:       lagScores,
+		Players:         players,
+		PlayerColors:    pc,
+		Phase:           phase,
+		CurrentDealer:   curDealerID,
+		CutCard:         cutCard,
+		Crib:            cribCards,
+		BlockingPlayers: bp,
+		Hands:           h,
+		PeggedCards:     p,
+		Actions:         pas,
+	}
+
+	return game, nil
+}
+
+func populateScores(
+	scoreBlue, scoreRed, scoreGreen,
+	lagScoreBlue, lagScoreRed, lagScoreGreen uint8,
+) (cur, lag map[model.PlayerColor]int) {
+	curScores := make(map[model.PlayerColor]int, 3)
+	lagScores := make(map[model.PlayerColor]int, 3)
+	if scoreBlue > 0 {
+		curScores[model.Blue] = int(scoreBlue)
+		lagScores[model.Blue] = int(lagScoreBlue)
+	}
+	if scoreRed > 0 {
+		curScores[model.Red] = int(scoreRed)
+		lagScores[model.Red] = int(lagScoreRed)
+	}
+	if scoreGreen > 0 {
+		curScores[model.Green] = int(scoreGreen)
+		lagScores[model.Green] = int(lagScoreGreen)
+	}
+
+	return curScores, lagScores
+}
+
+func addInPopulatedColor(
+	curScores, lagScores map[model.PlayerColor]int,
+	pc map[model.PlayerID]model.PlayerColor,
+) {
+	// if we know what color the players are, but we don't have point entries
+	// for those colors in the scores maps, add zeros
+	for _, color := range pc {
+		if _, ok := curScores[color]; !ok {
+			curScores[color] = 0
+		}
+		if _, ok := lagScores[color]; !ok {
+			lagScores[color] = 0
+		}
+	}
+}
+
+func getCribCards(cribCardInt int32) []model.Card {
+	var cribCards []model.Card
+	var cci int8
+	for i := uint(0); i < 4; i++ {
+		cci = int8(cribCardInt >> (8 * i))
+		c, err := model.NewCardFromTinyInt(cci)
+		if err != nil {
+			// If we've errored here, we assume it just means the card isn't set
+			continue
+		}
+		cribCards = append(cribCards, c)
+	}
+	return cribCards
+}
+
+func serializeCribCards(crib []model.Card) int32 {
+	val := int32(0)
+	for i := uint(0); i < 4; i++ {
+		ti := int8(0x34)
```

Comment:
don't love magic constants... do we already have 52 defined as max cards somewhere?